### PR TITLE
add test for conversion.go

### PR DESCRIPTION
**Pull request by adrianwg23**
Super simple PR for adding tests to `conversion.go` in the gogo package

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pkg/util/gogo/conversion_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+				Seconds: 1000000,
+				Nanos:   100000,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			got := DurationToProtoDuration(tt.gogo)
+			if (got == nil && tt.expectedValue != nil) || (got != nil && tt.expectedValue == nil) {
```
Comment (adiprerepa) [inline pkg/util/gogo/conversion_test.go:91]: remove the parenthesis from the if

File: pkg/util/gogo/conversion_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+				t.Errorf("%s: got: %v, expected: %v", tt.desc, got, tt.expectedValue)
+			} else if (got != nil && tt.expectedValue != nil) && (got.Seconds != tt.expectedValue.Seconds || got.Nanos != tt.expectedValue.Nanos) {
```
Comment (adiprerepa) [inline pkg/util/gogo/conversion_test.go:93]: why cant you just use one `reflect.DeepEqual()`?

**Review by adiprerepa — COMMENTED** [review]
general nits

File: pkg/util/gogo/conversion_test.go (hunk 2 above)
Comment (adrianwg23) [reply pkg/util/gogo/conversion_test.go:93]: still learning go! good catch on that.

File: pkg/util/gogo/conversion_test.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+				Value: false,
+			},
+			expectedValue: proto.BoolFalse,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			got := BoolToProtoBool(tt.gogo)
+			if got != tt.expectedValue {
```
Comment (adiprerepa) [inline pkg/util/gogo/conversion_test.go:58]: should be very easy to do something like:

```
if got := BoolToProtoBool(tt.gogo); got != tt.expectedValue {
}
```

File: pkg/util/gogo/conversion_test.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+				Seconds: 1000000,
+				Nanos:   100000,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			got := DurationToProtoDuration(tt.gogo)
+			if !reflect.DeepEqual(got, tt.expectedValue) {
```
Comment (adiprerepa) [inline pkg/util/gogo/conversion_test.go:92]: same comment from above

**Review by adiprerepa — APPROVED** [review]
LGTM after that nit

**Review by howardjohn — APPROVED** [review]

**Pull request merged**